### ISSUE TITLE
CSP solution

### ISSUE THREAD

**Issue by szepeviktor** [owncloud/core#17758]
New Firefox (41) stops function generation:
```js
func = new Function("n", code);
```
Here: https://github.com/owncloud/core/blob/master/core/js/l10n.js#L119

I think you'd like to keep `unsafe-eval` so the only solution is to rewrite that part of Javascript.

All the best wishes to you!

**Comment by szepeviktor** [comment]
```
Content Security Policy: The page's settings blocked the loading of a resource at self ("script-src https://---.net 'unsafe-eval'").
```

**Comment by szepeviktor** [comment]
It's funny that I've found the location in Chrome.

**Comment by LukasReschke** [comment]
> I think you'd like to keep unsafe-eval so the only solution is to rewrite that part of Javascript.

`unsafe-eval` is already set per default as you can see at https://github.com/owncloud/core/blob/d20e2002a6de9d472d29fb190867970a6f6b3b22/lib/public/appframework/http/contentsecuritypolicy.php#L43-L48, where exactly do you see that error and on what ownCloud version?

**Comment by szepeviktor** [comment]
Of course on the latest stable: 8.1.0.8.
I see it on every page which includes `l10n.js`.
You can get a new Firefox here: https://www.mozilla.org/en-US/firefox/developer/all/

**Comment by LukasReschke** [comment]
Thanks. This seems to be a Firefox bug. I will investigate this one.

**Comment by szepeviktor** [comment]
I think they strengthen `unsafe-eval`.

**Comment by oparoz** [comment]
Isn't this an old issue? I've had that for months on Firefox.

**Comment by szepeviktor** [comment]
Yes it is quiet old, but unsolved.

**Comment by MorrisJobke** [comment]
@LukasReschke Is this a duplicate of #10314?

**Comment by szepeviktor** [comment]
@MorrisJobke Could you please copy my technical reports to the original ticket after closing my tickets?

**Comment by szepeviktor** [comment]
Chrome 44 says:
```
Refused to load the image 'data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==' because it violates the following Content Security Policy directive: "img-src 'self'".
```

**Comment by MorrisJobke** [comment]
> Refused to load the image 'data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==' because it violates the following Content Security Policy directive: "img-src 'self'".

This was fixed by #19426 in 8.2 and 8.1.4 - other comments are copied to #10314 -> closing this one
